pkg/gormutil/logger.go

```diff
@@ -0,0 +1,138 @@
+package gormutil
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/iver-wharf/wharf-core/internal/traceutil"
+	"github.com/iver-wharf/wharf-core/pkg/logger"
+	"gorm.io/gorm"
+	gormlogger "gorm.io/gorm/logger"
+)
+
+// LoggerConfig holds configuration for the GORM logging integration. Many
+// configuration values of in the gorm.io/gorm/logger.Config will you also find
+// in here.
+type LoggerConfig struct {
+	// Logger is the logger implementation used when logging. This defaults to
+	// a new scoped logger with the scope "GORM".
+	Logger logger.Logger
+	// AlsoUseGORMLogLevel sets wether to honor GORM's own logging levels.
+	//
+	// If set to false (which is the default) then the logging level
+	// configuration from the wharf-core logging library will be the only one
+	// filtering logs.
+	AlsoUseGORMLogLevel bool
+	// IgnoreRecordNotFoundError will omit any "RecordNotFound" errors if set
+	// to true.
+	IgnoreRecordNotFoundError bool
+	// SlowThreshold sets at which minimal time duration is considered a slow
+	// SQL operation. If an operation takes more than this time to complete then
+	// a warning log message will be submitted.
+	SlowThreshold time.Duration
```

Comment:
nit: I prefer this wording more. Along with some additional info of what 0 does.
```suggestion
	// SlowThreshold sets what duration is considered a slow SQL operation.
	// If an operation takes longer than this to complete then a warning log
	// message will be emitted.
	// 
	// Set to 0 to disable.
	SlowThreshold time.Duration
```